runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -344,29 +344,33 @@ void J9::RecognizedCallTransformer::process_java_lang_StrictMath_and_Math_sqrt(T
 Transform an Unsafe atomic call to diamonds with equivalent semantics
 
                           yes
-isObjectNull ------------------------------------------>
-    |                                                  |
-    | no                                               |
-    |                     yes                          |
-isNotLowTagged ---------------------------------------->
-    |                                                  |
-    |  no                                              |
-    |           no                                     |
-isFinal ---------------------->                        |
-    |                         |                        |
-    | yes                     |                        |
-    |                         |                        |
-call the                calculate address      calculate address
-original method         for static field       for instance field
-    |                         |                and absolute address
-    |                         |                        |
-    |                         |________________________|
-    |                                     |
-    |                         xcall atomic method helper
-    |                                     |
-    |_____________________________________|
-                    |
-      program after the original call
+isObjectNull -------------------------------------------->
+    |                                                    |
+    | no                                                 |
+    |                     yes                            |
+isNotLowTagged ------------------------------------------>
+    |                                                    |
+    |  no                                                |
+    |           no                                       |
+isFinal ------------------------>                        |
+    |                           |                        |
+    | yes                       |                        |
+jitReportFinalFieldModified     |                        |
```

Comment:
I assume the case when `isFinal` is true is just an improvement to the documentation rather than any change in logic as such ?

Reply:
Yes, it's just to update the documentation for `isFinal`. My understanding of the existing implementation on it is that the old diagram [1] doesn't reflect what the code (`createConditionalAlternatePath`) [2] does. The merge block for `isFinal` is the block that calculates static address since the `treetop->getEnclosingBlock()` is split later [3]. 

[1]
https://github.com/eclipse-openj9/openj9/blob/f665453387d2b03bc617bfa4b56693e430c2d1c2/runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp#L354-L369

[2]
https://github.com/eclipse-openj9/openj9/blob/f665453387d2b03bc617bfa4b56693e430c2d1c2/runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp#L515-L522

[3]
https://github.com/eclipse-openj9/openj9/blob/f665453387d2b03bc617bfa4b56693e430c2d1c2/runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp#L551